qiskit_aer/quantum_info/states/aer_densitymatrix.py

```diff
@@ -0,0 +1,354 @@
+        return self._result
+
+    def metadata(self):
+        """Return result metadata of an operation that executed lastly."""
+        if self._last_result() is None:
+            raise AerError('AerState was not used and metdata does not exist.')
+        return self._last_result()['metadata']
+
+    def __copy__(self):
+        return copy.deepcopy(self)
+
+    def __deepcopy__(self, _memo=None):
+        ret = AerDensityMatrix(self._data.copy(), **self._configs)
+        ret._op_shape = copy.deepcopy(self._op_shape)
+        ret._rng_generator = copy.deepcopy(self._rng_generator)
+        return ret
+
+    def conjugate(self):
+        return AerDensityMatrix(np.conj(self._data), dims=self.dims())
+
+    def tensor(self, other):
+        """Return the tensor product state self ⊗ other.
+        Args:
+            other (AerDensityMatrix): a quantum state object.
+        Returns:
+            AerDensityMatrix: the tensor product operator self ⊗ other.
+        Raises:
+            QiskitError: if other is not a quantum state.
+        """
+        if not isinstance(other, AerDensityMatrix):
+            other = AerDensityMatrix(other)
+        ret = copy.copy(self)
+        ret._data = np.kron(self._data, other._data)
+        ret._op_shape = self._op_shape.tensor(other._op_shape)
+        return ret
+
+    def expand(self, other):
+        """Return the tensor product state other ⊗ self.
+        Args:
+            other (AerDensityMatrix): a quantum state object.
+        Returns:
+            AerDensityMatrix: the tensor product state other ⊗ self.
+        Raises:
+            QiskitError: if other is not a quantum state.
+        """
+        if not isinstance(other, AerDensityMatrix):
+            other = AerDensityMatrix(other)
+        ret = copy.copy(self)
+        ret._data = np.kron(other._data, self._data)
+        ret._op_shape = self._op_shape.expand(other._op_shape)
+        return ret
+
+    def _add(self, other):
+        """Return the linear combination self + other.
+        Args:
+            other (AerDensityMatrix): a quantum state object.
+        Returns:
+            AerDensityMatrix: the linear combination self + other.
+        Raises:
+            QiskitError: if other is not a quantum state, or has
+                         incompatible dimensions.
+        """
+        if not isinstance(other, AerDensityMatrix):
+            other = AerDensityMatrix(other)
+        self._op_shape._validate_add(other._op_shape)
+        ret = copy.copy(self)
+        ret._data = self.data + other.data
+        return ret
+
+    def sample_memory(self, shots, qargs=None):
+        if qargs is None:
+            qubits = np.arange(self._aer_state.num_qubits)
+        else:
+            qubits = np.array(qargs)
+
+        aer_state = AerState(**self._aer_state.configuration())
+        aer_state.initialize(self._data, copy=False)
+        samples = aer_state.sample_memory(qubits, shots)
+        aer_state.close()
```

Comment:
```suggestion
        self._aer_state.close()

        self._aer_state.renew()
        self._aer_state.initialize(self._data, copy=False)
        samples = self._aer_state.sample_memory(qubits, shots)
        self._data = self._aer_state.move_to_ndarray()
```

To keep a random generator in `AerState`, `sample_memory` needs the same `AerState` instance.

Reply:
Thank you. I will correct it according to your suggestion and #1663.